{"nodes":[{"pos":[775,874],"content":"Provides data for the <bpt id=\"p1\">&lt;xref href=\"System.Web.UI.WebControls.LinqDataSource.Deleting\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> event.","needQuote":true,"needEscape":true,"source":"Provides data for the <xref href=\"System.Web.UI.WebControls.LinqDataSource.Deleting\"></xref> event."},{"pos":[887,2906],"content":"The LinqDataSourceDeleteEventArgs object is passed to any handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event. The <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject%2A> property contains the data that will be deleted.       If the object that represents the data source throws a validation exception before it deletes the data, the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A> property contains an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class. You can retrieve all the validation exceptions through the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property. If no validation exception is thrown, the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A> property contains `null`. If you handle the validation exceptions and do not want the exception to be re-thrown, set the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled%2A> property to `true`.       You create an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event to validate the data, to examine validation errors from the data class, or to cancel the delete operation. You cancel the delete operation by setting the <xref:System.ComponentModel.CancelEventArgs.Cancel%2A> property to `true`.       By default, the <xref:System.Web.UI.WebControls.LinqDataSource> control stores the original values from the data source in view state in the Web page, except those whose <xref:System.Data.Linq.Mapping.ColumnAttribute> attribute is marked as `UpdateCheck.Never`. LINQ to SQL automatically checks the integrity of the data before deleting the data. It does this by comparing the current values in the data source with the original values that are stored in view state. You can perform additional data validation by creating a handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event.","needQuote":false,"needEscape":true,"extradata":"MT","nodes":[{"content":"The LinqDataSourceDeleteEventArgs object is passed to any handler for the &lt;xref:System.Web.UI.WebControls.LinqDataSource.Deleting&gt; event.","pos":[0,137],"source":"The LinqDataSourceDeleteEventArgs object is passed to any handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event."},{"content":"The &lt;xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject%2A&gt; property contains the data that will be deleted.","pos":[138,271],"source":" The <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject%2A> property contains the data that will be deleted."},{"content":"If the object that represents the data source throws a validation exception before it deletes the data, the &lt;xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A&gt; property contains an instance of the &lt;xref:System.Web.UI.WebControls.LinqDataSourceValidationException&gt; class.","pos":[278,572],"source":"       If the object that represents the data source throws a validation exception before it deletes the data, the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A> property contains an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class."},{"content":"You can retrieve all the validation exceptions through the &lt;xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A&gt; property.","pos":[573,727],"source":" You can retrieve all the validation exceptions through the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property."},{"content":"If no validation exception is thrown, the &lt;xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A&gt; property contains <ph id=\"ph1\">`null`</ph>.","pos":[728,871],"source":" If no validation exception is thrown, the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A> property contains `null`."},{"content":"If you handle the validation exceptions and do not want the exception to be re-thrown, set the &lt;xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled%2A&gt; property to <ph id=\"ph1\">`true`</ph>.","pos":[872,1069],"source":" If you handle the validation exceptions and do not want the exception to be re-thrown, set the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled%2A> property to `true`."},{"content":"You create an event handler for the &lt;xref:System.Web.UI.WebControls.LinqDataSource.Deleting&gt; event to validate the data, to examine validation errors from the data class, or to cancel the delete operation.","pos":[1076,1281],"source":"       You create an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event to validate the data, to examine validation errors from the data class, or to cancel the delete operation."},{"content":"You cancel the delete operation by setting the &lt;xref:System.ComponentModel.CancelEventArgs.Cancel%2A&gt; property to <ph id=\"ph1\">`true`</ph>.","pos":[1282,1403],"source":" You cancel the delete operation by setting the <xref:System.ComponentModel.CancelEventArgs.Cancel%2A> property to `true`."},{"content":"By default, the &lt;xref:System.Web.UI.WebControls.LinqDataSource&gt; control stores the original values from the data source in view state in the Web page, except those whose &lt;xref:System.Data.Linq.Mapping.ColumnAttribute&gt; attribute is marked as <ph id=\"ph1\">`UpdateCheck.Never`</ph>.","pos":[1410,1671],"source":"       By default, the <xref:System.Web.UI.WebControls.LinqDataSource> control stores the original values from the data source in view state in the Web page, except those whose <xref:System.Data.Linq.Mapping.ColumnAttribute> attribute is marked as `UpdateCheck.Never`."},{"content":"LINQ to SQL automatically checks the integrity of the data before deleting the data.","pos":[1672,1756]},{"content":"It does this by comparing the current values in the data source with the original values that are stored in view state.","pos":[1757,1876]},{"content":"You can perform additional data validation by creating a handler for the &lt;xref:System.Web.UI.WebControls.LinqDataSource.Deleting&gt; event.","pos":[1877,2013],"source":" You can perform additional data validation by creating a handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event."}]},{"pos":[4356,4473],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\"></xref> class."},{"pos":[4645,4785],"content":"An object of the type specified in the <xref:System.Web.UI.WebControls.LinqDataSource.TableName*> property that contains the data to delete.","needQuote":true,"needEscape":true,"nodes":[{"content":"An object of the type specified in the &lt;xref:System.Web.UI.WebControls.LinqDataSource.TableName*&gt; property that contains the data to delete.","pos":[0,140],"source":"An object of the type specified in the <xref:System.Web.UI.WebControls.LinqDataSource.TableName*> property that contains the data to delete."}]},{"pos":[5619,5765],"content":"Initializes a new instance of the <bpt id=\"p1\">&lt;xref href=\"System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class with the specified exception.","needQuote":true,"needEscape":true,"source":"Initializes a new instance of the <xref href=\"System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\"></xref> class with the specified exception."},{"pos":[6026,6089],"content":"The exception that occurred when validating the data to delete.","needQuote":true,"needEscape":true,"nodes":[{"content":"The exception that occurred when validating the data to delete.","pos":[0,63]}]},{"pos":[6639,6737],"content":"Gets the exception that was thrown while the data was being validated before the delete operation.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the exception that was thrown while the data was being validated before the delete operation.","pos":[0,98]}]},{"pos":[6749,7032],"content":"If a validation exception is thrown before the delete operation is performed, the exception is stored in the Exception property. In an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event, you can retrieve the exception through the Exception property.","needQuote":true,"needEscape":true,"extradata":"MT","nodes":[{"content":"If a validation exception is thrown before the delete operation is performed, the exception is stored in the Exception property. In an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event, you can retrieve the exception through the Exception property.","pos":[0,283],"nodes":[{"content":"If a validation exception is thrown before the delete operation is performed, the exception is stored in the Exception property.","pos":[0,128]},{"content":"In an event handler for the &lt;xref:System.Web.UI.WebControls.LinqDataSource.Deleting&gt; event, you can retrieve the exception through the Exception property.","pos":[129,283],"source":" In an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event, you can retrieve the exception through the Exception property."}]}]},{"pos":[7245,7354],"content":"The exception, if an error occurred; otherwise, <bpt id=\"p1\">&lt;xref uid=\"langword_csharp_null\" name=\"null\" href=\"\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept>.","needQuote":true,"needEscape":true,"source":"The exception, if an error occurred; otherwise, <xref uid=\"langword_csharp_null\" name=\"null\" href=\"\"></xref>."},{"pos":[7943,8052],"content":"Gets or sets a value that indicates whether the exception was handled and that it should not be thrown again.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets or sets a value that indicates whether the exception was handled and that it should not be thrown again.","pos":[0,109]}]},{"pos":[8064,8494],"content":"You can create an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event to examine validation exceptions that were thrown before the delete operation. If you handle the exception and do not want it to be thrown again, set the ExceptionHandled property to `true`. If you do not set the ExceptionHandled property to `true`, the exception will be propagated to the next event handler in the call stack.","needQuote":true,"needEscape":true,"extradata":"MT","nodes":[{"content":"You can create an event handler for the &lt;xref:System.Web.UI.WebControls.LinqDataSource.Deleting&gt; event to examine validation exceptions that were thrown before the delete operation.","pos":[0,181],"source":"You can create an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event to examine validation exceptions that were thrown before the delete operation."},{"content":"If you handle the exception and do not want it to be thrown again, set the ExceptionHandled property to <ph id=\"ph1\">`true`</ph>.","pos":[182,293],"source":" If you handle the exception and do not want it to be thrown again, set the ExceptionHandled property to `true`."},{"content":"If you do not set the ExceptionHandled property to <ph id=\"ph1\">`true`</ph>, the exception will be propagated to the next event handler in the call stack.","pos":[294,430],"source":" If you do not set the ExceptionHandled property to `true`, the exception will be propagated to the next event handler in the call stack."}]},{"pos":[8619,8784],"content":"<bpt id=\"p1\">&lt;xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> if the exception was handled; otherwise, <bpt id=\"p2\">&lt;xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"&gt;</bpt><ept id=\"p2\">&lt;/xref&gt;</ept>.","needQuote":true,"needEscape":true,"source":"<xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref> if the exception was handled; otherwise, <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>."},{"pos":[9370,9421],"content":"Gets the object that represents the data to delete.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the object that represents the data to delete.","pos":[0,51]}]},{"pos":[9433,9782],"content":"You can use the OriginalObject property to interact with the data before it is deleted. You can validate the data, or you can cancel the event by setting the <xref:System.ComponentModel.CancelEventArgs.Cancel%2A> property to `true`. You can modify which record is deleted by changing the value or values of the object in the OriginalObject property.","needQuote":true,"needEscape":true,"extradata":"MT","nodes":[{"content":"You can use the OriginalObject property to interact with the data before it is deleted.","pos":[0,87]},{"content":"You can validate the data, or you can cancel the event by setting the &lt;xref:System.ComponentModel.CancelEventArgs.Cancel%2A&gt; property to <ph id=\"ph1\">`true`</ph>.","pos":[88,232],"source":" You can validate the data, or you can cancel the event by setting the <xref:System.ComponentModel.CancelEventArgs.Cancel%2A> property to `true`."},{"content":"You can modify which record is deleted by changing the value or values of the object in the OriginalObject property.","pos":[233,349]}]},{"pos":[10510,10650],"content":"An object of the type specified in the <xref:System.Web.UI.WebControls.LinqDataSource.TableName*> property that contains the data to delete.","needQuote":true,"needEscape":true,"nodes":[{"content":"An object of the type specified in the &lt;xref:System.Web.UI.WebControls.LinqDataSource.TableName*&gt; property that contains the data to delete.","pos":[0,140],"source":"An object of the type specified in the <xref:System.Web.UI.WebControls.LinqDataSource.TableName*> property that contains the data to delete."}]}],"content":"### YamlMime:ManagedReference\nitems:\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  id: LinqDataSourceDeleteEventArgs\n  children:\n  - System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor(System.Object)\n  - System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor(System.Web.UI.WebControls.LinqDataSourceValidationException)\n  - System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception\n  - System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled\n  - System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject\n  langs:\n  - csharp\n  name: LinqDataSourceDeleteEventArgs\n  nameWithType: LinqDataSourceDeleteEventArgs\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  type: Class\n  summary: Provides data for the <xref href=\"System.Web.UI.WebControls.LinqDataSource.Deleting\"></xref> event.\n  remarks: \"The LinqDataSourceDeleteEventArgs object is passed to any handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event. The <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject%2A> property contains the data that will be deleted.  \\n  \\n If the object that represents the data source throws a validation exception before it deletes the data, the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A> property contains an instance of the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException> class. You can retrieve all the validation exceptions through the <xref:System.Web.UI.WebControls.LinqDataSourceValidationException.InnerExceptions%2A> property. If no validation exception is thrown, the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception%2A> property contains `null`. If you handle the validation exceptions and do not want the exception to be re-thrown, set the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled%2A> property to `true`.  \\n  \\n You create an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event to validate the data, to examine validation errors from the data class, or to cancel the delete operation. You cancel the delete operation by setting the <xref:System.ComponentModel.CancelEventArgs.Cancel%2A> property to `true`.  \\n  \\n By default, the <xref:System.Web.UI.WebControls.LinqDataSource> control stores the original values from the data source in view state in the Web page, except those whose <xref:System.Data.Linq.Mapping.ColumnAttribute> attribute is marked as `UpdateCheck.Never`. LINQ to SQL automatically checks the integrity of the data before deleting the data. It does this by comparing the current values in the data source with the original values that are stored in view state. You can perform additional data validation by creating a handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event.\"\n  example:\n  - \"The following example shows how to cancel the delete operation based on a property in the <xref:System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject%2A> property and a value from the Web page. In the example, users must select a <xref:System.Web.UI.WebControls.CheckBox> control to confirm that they want to delete a product record when its `OnSale` property is set to `true`.  \\n  \\n [!code-vb[System.Web.UI.WebControls.LinqDataSource.Deleting#1](~/add/codesnippet/visualbasic/t-system.web.ui.webcontr_255_1.vb)]\\n [!code-cs[System.Web.UI.WebControls.LinqDataSource.Deleting#1](~/add/codesnippet/csharp/t-system.web.ui.webcontr_255_1.cs)]\"\n  syntax:\n    content: 'public class LinqDataSourceDeleteEventArgs : System.ComponentModel.CancelEventArgs'\n  inheritance:\n  - System.ComponentModel.CancelEventArgs\n  implements: []\n  inheritedMembers: []\n  platform:\n  - net462\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor(System.Object)\n  id: '#ctor(System.Object)'\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  langs:\n  - csharp\n  name: LinqDataSourceDeleteEventArgs(Object)\n  nameWithType: LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(Object)\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(Object)\n  type: Constructor\n  assemblies:\n  - System.Web.Extensions\n  namespace: System.Web.UI.WebControls\n  summary: Initializes a new instance of the <xref href=\"System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\"></xref> class.\n  syntax:\n    content: public LinqDataSourceDeleteEventArgs (object originalObject);\n    parameters:\n    - id: originalObject\n      type: System.Object\n      description: An object of the type specified in the <xref:System.Web.UI.WebControls.LinqDataSource.TableName*> property that contains the data to delete.\n  overload: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor(System.Web.UI.WebControls.LinqDataSourceValidationException)\n  id: '#ctor(System.Web.UI.WebControls.LinqDataSourceValidationException)'\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  langs:\n  - csharp\n  name: LinqDataSourceDeleteEventArgs(LinqDataSourceValidationException)\n  nameWithType: LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(LinqDataSourceValidationException)\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(LinqDataSourceValidationException)\n  type: Constructor\n  assemblies:\n  - System.Web.Extensions\n  namespace: System.Web.UI.WebControls\n  summary: Initializes a new instance of the <xref href=\"System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\"></xref> class with the specified exception.\n  syntax:\n    content: public LinqDataSourceDeleteEventArgs (System.Web.UI.WebControls.LinqDataSourceValidationException exception);\n    parameters:\n    - id: exception\n      type: System.Web.UI.WebControls.LinqDataSourceValidationException\n      description: The exception that occurred when validating the data to delete.\n  overload: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception\n  id: Exception\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  langs:\n  - csharp\n  name: Exception\n  nameWithType: LinqDataSourceDeleteEventArgs.Exception\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception\n  type: Property\n  assemblies:\n  - System.Web.Extensions\n  namespace: System.Web.UI.WebControls\n  summary: Gets the exception that was thrown while the data was being validated before the delete operation.\n  remarks: If a validation exception is thrown before the delete operation is performed, the exception is stored in the Exception property. In an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event, you can retrieve the exception through the Exception property.\n  syntax:\n    content: public System.Web.UI.WebControls.LinqDataSourceValidationException Exception { get; }\n    return:\n      type: System.Web.UI.WebControls.LinqDataSourceValidationException\n      description: The exception, if an error occurred; otherwise, <xref uid=\"langword_csharp_null\" name=\"null\" href=\"\"></xref>.\n  overload: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled\n  id: ExceptionHandled\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  langs:\n  - csharp\n  name: ExceptionHandled\n  nameWithType: LinqDataSourceDeleteEventArgs.ExceptionHandled\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled\n  type: Property\n  assemblies:\n  - System.Web.Extensions\n  namespace: System.Web.UI.WebControls\n  summary: Gets or sets a value that indicates whether the exception was handled and that it should not be thrown again.\n  remarks: You can create an event handler for the <xref:System.Web.UI.WebControls.LinqDataSource.Deleting> event to examine validation exceptions that were thrown before the delete operation. If you handle the exception and do not want it to be thrown again, set the ExceptionHandled property to `true`. If you do not set the ExceptionHandled property to `true`, the exception will be propagated to the next event handler in the call stack.\n  syntax:\n    content: public bool ExceptionHandled { get; set; }\n    return:\n      type: System.Boolean\n      description: <xref uid=\"langword_csharp_true\" name=\"true\" href=\"\"></xref> if the exception was handled; otherwise, <xref uid=\"langword_csharp_false\" name=\"false\" href=\"\"></xref>.\n  overload: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject\n  id: OriginalObject\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  langs:\n  - csharp\n  name: OriginalObject\n  nameWithType: LinqDataSourceDeleteEventArgs.OriginalObject\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject\n  type: Property\n  assemblies:\n  - System.Web.Extensions\n  namespace: System.Web.UI.WebControls\n  summary: Gets the object that represents the data to delete.\n  remarks: You can use the OriginalObject property to interact with the data before it is deleted. You can validate the data, or you can cancel the event by setting the <xref:System.ComponentModel.CancelEventArgs.Cancel%2A> property to `true`. You can modify which record is deleted by changing the value or values of the object in the OriginalObject property.\n  example:\n  - \"The following example shows how to cancel the delete operation based on a property in the OriginalObject property and a value from the Web page. In the example, users must select a <xref:System.Web.UI.WebControls.CheckBox> control to confirm that they want to delete a product record when its `OnSale` property is set to `true`.  \\n  \\n [!code-vb[System.Web.UI.WebControls.LinqDataSource.Deleting#1](~/add/codesnippet/visualbasic/p-system.web.ui.webcontr_196_1.vb)]\\n [!code-cs[System.Web.UI.WebControls.LinqDataSource.Deleting#1](~/add/codesnippet/csharp/p-system.web.ui.webcontr_196_1.cs)]\"\n  syntax:\n    content: public object OriginalObject { get; }\n    return:\n      type: System.Object\n      description: An object of the type specified in the <xref:System.Web.UI.WebControls.LinqDataSource.TableName*> property that contains the data to delete.\n  overload: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject*\n  exceptions: []\n  platform:\n  - net462\nreferences:\n- uid: System.ComponentModel.CancelEventArgs\n  isExternal: false\n  name: System.ComponentModel.CancelEventArgs\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor(System.Object)\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: LinqDataSourceDeleteEventArgs(Object)\n  nameWithType: LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(Object)\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(Object)\n- uid: System.Object\n  parent: System\n  isExternal: true\n  name: Object\n  nameWithType: Object\n  fullName: System.Object\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor(System.Web.UI.WebControls.LinqDataSourceValidationException)\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: LinqDataSourceDeleteEventArgs(LinqDataSourceValidationException)\n  nameWithType: LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(LinqDataSourceValidationException)\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs(LinqDataSourceValidationException)\n- uid: System.Web.UI.WebControls.LinqDataSourceValidationException\n  parent: System.Web.UI.WebControls\n  isExternal: false\n  name: LinqDataSourceValidationException\n  nameWithType: LinqDataSourceValidationException\n  fullName: System.Web.UI.WebControls.LinqDataSourceValidationException\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: Exception\n  nameWithType: LinqDataSourceDeleteEventArgs.Exception\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: ExceptionHandled\n  nameWithType: LinqDataSourceDeleteEventArgs.ExceptionHandled\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled\n- uid: System.Boolean\n  parent: System\n  isExternal: true\n  name: Boolean\n  nameWithType: Boolean\n  fullName: System.Boolean\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: OriginalObject\n  nameWithType: LinqDataSourceDeleteEventArgs.OriginalObject\n  fullName: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.#ctor*\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: LinqDataSourceDeleteEventArgs\n  nameWithType: LinqDataSourceDeleteEventArgs.LinqDataSourceDeleteEventArgs\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.Exception*\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: Exception\n  nameWithType: LinqDataSourceDeleteEventArgs.Exception\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.ExceptionHandled*\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: ExceptionHandled\n  nameWithType: LinqDataSourceDeleteEventArgs.ExceptionHandled\n- uid: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs.OriginalObject*\n  parent: System.Web.UI.WebControls.LinqDataSourceDeleteEventArgs\n  isExternal: false\n  name: OriginalObject\n  nameWithType: LinqDataSourceDeleteEventArgs.OriginalObject\n"}